editor/src/clj/editor/command_requests.clj

```diff
@@ -0,0 +1,270 @@
+   {:ui-handler :show-console
+    :help "Show the Console tab."}
+
+   :show-curve-editor
+   {:ui-handler :show-curve-editor
+    :help "Show the Curve Editor tab."}
+
+   :support-forum
+   {:ui-handler :support-forum
+    :help "Open the Defold Support Forum in a web browser."}
+
+   :toggle-pane-bottom
+   {:ui-handler :toggle-pane-bottom
+    :help "Toggle visibility of the bottom editor pane."}
+
+   :toggle-pane-left
+   {:ui-handler :toggle-pane-left
+    :help "Toggle visibility of the left editor pane."}
+
+   :toggle-pane-right
+   {:ui-handler :toggle-pane-right
+    :help "Toggle visibility of the right editor pane."}})
+
+(defonce ^:private help-response
+  (let [longest-command-length
+        (reduce (fn [result [command]]
+                  (max result (count (name command))))
+                Long/MIN_VALUE
+                supported-commands)
+
+        command-format-string
+        (str "  %-" (+ longest-command-length 2) "s : %s")
+
+        help-json-string
+        (str "{\n"
+             (->> supported-commands
+                  (sort-by key)
+                  (map (fn [[command {:keys [help]}]]
+                         (format command-format-string
+                                 (str \" (name command) \")
+                                 (json/write-str help))))
+                  (string/join ",\n"))
+             "\n}\n")
+
+        utf8-bytes
+        (.getBytes help-json-string StandardCharsets/UTF_8)]
+    {:code 200
+     :headers {"Content-Type" "text/plain;charset=UTF-8"
+               "Content-Length" (str (count utf8-bytes))}
+     :body utf8-bytes}))
+
+(defonce ^:private malformed-request-response
+  {:code 400
+   :body "400 Bad Request"})
+
+(defonce ^:private command-not-supported-response
+  {:code 404
+   :body "404 Not Found"})
+
+(defonce ^:private command-not-enabled-response
+  {:code 405
+   :body "405 Method Not Allowed"})
+
+(defonce ^:private internal-server-error-response
+  {:code 500
+   :body "500 Internal Server Error"})
+
+(defonce ^:private command-accepted-response
+  {:code 202
+   :body "202 Accepted"})
+
+(defn- parse-command [request]
+  (let [url-string (:url request)]
+    (case url-string
+      ("/command" "/command/") ::api
+      (try
+        (keyword (subs url-string 9))
+        (catch Exception error
+          (log/error :msg "Failed to parse command request"
+                     :request request
+                     :exception error)
+          nil)))))
+
+(defn- resolve-ui-handler-ctx [ui-node command user-data]
+  {:pre [(ui/node? ui-node)
+         (keyword? command)
+         (map? user-data)]}
+  (ui/run-now
+    (let [command-contexts (ui/node-contexts ui-node true)]
+      (ui/resolve-handler-ctx command-contexts command user-data))))
+
+(defn make-request-handler [ui-node render-reload-progress!]
+  {:pre [(ui/node? ui-node)
+         (fn? render-reload-progress!)]}
+  (fn request-handler [request]
+    (when (= "GET" (:method request))
```

Comment:
I think we should not accept `GET` for command execution because `GET` implies read-only semantics that does not have side effects. Maybe `POST` will work better?

Reply:
That makes sense. I'll change it to POST!